lib/widgets/received-requests.dart

```diff
@@ -46,31 +53,76 @@ class SingleRequest extends StatelessWidget {
     return new ListTile(
       title: Text(request['name']),
       subtitle: Text(request['email']),
-      trailing: IconRow(),
+      trailing: IconRow(request),
     );
   }
 }
 
-//IconRow is a widget that renders the accept button and deny button for each request widget. 
+//IconRow is a widget that renders the accept button and deny button for each request widget.
 class IconRow extends StatelessWidget {
+  final request;
+  IconRow(this.request);
+
   @override
   Widget build(BuildContext ctx) {
     return new Container(
-      child : Row(
+        child: Row(
       mainAxisSize: MainAxisSize.min,
       children: <Widget>[
         IconButton(
             icon: Icon(Icons.check),
-            onPressed: () {
-              print("Accepted Request!");
-            }),
+            onPressed: () => {acceptRequest(request)}),
         IconButton(
             icon: Icon(Icons.close),
-            onPressed: () {
-              print("Denied Request!");
-            })
+            onPressed: () => {removeRequest(request["id"])})
       ],
-    )
-    );
+    ));
   }
 }
+
+final databaseReference = FirebaseDatabase.instance.reference();
```

Comment:
Could you put this reference and the following functions inside the class?  I think they're global at the moment

Reply:
Yup! I'll get on that!